src/test_l3_dsHdmiIn.c

```diff
@@ -0,0 +1,1429 @@
+                     "false(i.e 0)\t"
+            "******************************************\n");
+    readInput(&select);
+    if(select < 0 || select > 1)
+    {
+       printf("\nInvalid audmix selected\n");
+    }
+    
+    topmost = (bool)select;
+
+    ret = dsHdmiInSelectPort(port , audmix , videoplanetype , topmost);
+    if(ret != dsERR_NONE)
+    {
+       UT_LOG_ERROR("Result: Failed to select the Hdmi input port:[%s]",ut_control_GetMapString(dsError_mapTable, ret));
+    } 
+           
+    UT_LOG_INFO("Result : dsHdmiInSelectPort: %s, portnumber: %d",
+                  ut_control_GetMapString(dsHdmiInPort_mapTable, port), port);
+    
+    UT_LOG_INFO("Out %s", __FUNCTION__);
+}
+
+/**
+* @brief This test to scale video.
+*
+* This test function is to scale video of selected HdmiInput port on platform.
+*
+* **Test Group ID:** 03@n
+* **Test Case ID:** 005@n
+*
+* **Test Procedure:**
+* Refer to Test specification documentation
+* [ds-hdmi-in_halSpec.md](../../docs/pages/ds-hdmi-in_halSpec.md)
+*/
+void test_l3_HdmiIn_scale_video(void)
+{
+    gTestID = 5;
+    UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+    dsError_t ret   = dsERR_NONE;
+    int32_t x = 0 , y = 0, width = 0 , height = 0;
+    int32_t select = 0;
+
+    printf(" \n================Please Select Inputs==================\n");
+    
+    printf("\n*******Enter the x coordinate to select*******\t"
+                     "Acceptable inputs are:\t"
+                     "Min is 0\t"
+                     "Max is based on the current resolution\t"
+              "**********************************************\n");
+   readInput(&select);
+   if(x < 0)
+   {
+      printf("\n invalid x coordinate selected \n");
+   }
+
+   printf("\n*******Enter the y coordinate to select*******\t"
+                     "Acceptable inputs are:\t"
+                     "Min is 0\t"
+                     "Max is based on the current resolution\t"
+             "**********************************************\n");
+   readInput(&select);
+   if(y < 0)
+   {
+      printf("\n invalid y coordinate selected \n");
+   }
+
+   printf("\n*******Enter the width to select***************\t"
+                     "Acceptable inputs are:\t"
+                     "Min is 0\t"
+                     "Max is based on the current resolution\t"
+            "**********************************************\n");
+   readInput(&select);
+   if(width < 0)
+   {
+      printf("\n invalid width selected \n");
+   }
+
+   printf("\n*******Enter the height to select**************\t"
+                     "Acceptable inputs are:\t"
+                     "Min is 0\t"
+                     "Max is based on the current resolution\t"
+             "**********************************************\n");
+   readInput(&select);
+   if(width < 0)
+   {
+      printf("\n invalid height selected \n");
+   }
+
+   ret = dsHdmiInScaleVideo(x, y, width, height);
+   if(ret != dsERR_NONE)
```

Comment:
Please consider the dsAudio L3 code and modify throughout the file

Reply:
done